spacy_llm/tests/tasks/test_summarization.py

```diff
@@ -0,0 +1,343 @@
+    path = {str((Path(__file__).parent / "templates" / "summarization.jinja2"))}
+
+    [components.llm.model]
+    @llm_models = "spacy.GPT-3-5.v1"
+    config = {{"temperature": 0}}
+    """
+
+
+@pytest.fixture
+def noop_config():
+    return """
+    [nlp]
+    lang = "en"
+    pipeline = ["llm"]
+    batch_size = 128
+
+    [components]
+
+    [components.llm]
+    factory = "llm"
+
+    [components.llm.task]
+    @llm_tasks = "spacy.Summarization.v1"
+
+    [components.llm.model]
+    @llm_models = "test.NoOpModel.v1"
+    """
+
+
+@pytest.fixture
+def example_text() -> str:
+    """Returns string to be used as example in tests."""
+    return (
+        "The atmosphere of Earth is the layer of gases, known collectively as air, retained by Earth's gravity "
+        "that surrounds the planet and forms its planetary atmosphere. The atmosphere of Earth creates pressure, "
+        "absorbs most meteoroids and ultraviolet solar radiation, warms the surface through heat retention "
+        "(greenhouse effect), allowing life and liquid water to exist on the Earth's surface, and reduces "
+        "temperature extremes between day and night (the diurnal temperature variation)."
+    )
+
+
+@pytest.mark.external
+@pytest.mark.skipif(has_openai_key is False, reason="OpenAI API key not available")
+@pytest.mark.parametrize(
+    "cfg_string",
+    [
+        "zeroshot_cfg_string",
+        "fewshot_cfg_string",
+        "ext_template_cfg_string",
+    ],
+)
+def test_summarization_config(cfg_string, request):
+    cfg_string = request.getfixturevalue(cfg_string)
+    orig_config = Config().from_str(cfg_string)
+    nlp = spacy.util.load_model_from_config(orig_config, auto_fill=True)
+    assert nlp.pipe_names == ["llm"]
+
+    # also test nlp config from a dict in add_pipe
+    component_cfg = dict(orig_config["components"]["llm"])
+    component_cfg.pop("factory")
+
+    nlp2 = spacy.blank("en")
+    nlp2.add_pipe("llm", config=component_cfg)
+    assert nlp2.pipe_names == ["llm"]
+
+    pipe = nlp.get_pipe("llm")
+    assert isinstance(pipe, LLMWrapper)
+    assert isinstance(pipe.task, LLMTask)
+
+
+@pytest.mark.external
+@pytest.mark.skipif(has_openai_key is False, reason="OpenAI API key not available")
+@pytest.mark.parametrize(
+    "cfg_string",
+    [
+        "zeroshot_cfg_string",
+        "fewshot_cfg_string",
+        "ext_template_cfg_string",
+    ],
+)
+def test_summarization_predict(cfg_string, example_text, request):
+    """Use OpenAI to get summarize text.
+    Note that this test may fail randomly, as the LLM's output is unguaranteed to be consistent/predictable
+    """
+    orig_cfg_string = cfg_string
+    cfg_string = request.getfixturevalue(cfg_string)
+    orig_config = Config().from_str(cfg_string)
+    nlp = spacy.util.load_model_from_config(orig_config, auto_fill=True)
+    doc = nlp(example_text)
+    # Check whether a non-empty summary was written and we are somewhat close to the desired upper length limit.
```

Comment:
"somewhat close" being implemented as "less than double" is a bit funny to me 😁

Reply:
I can see that :stuck_out_tongue: Lowered to 1.5. It was higher initially due to some incorrect examples used in the tests. Fixed in https://github.com/explosion/spacy-llm/pull/181/commits/861961ec9dce65001455b4228e000f9f3299ff0c.